### PR TITLE
docs: fix cni documentation

### PR DESCRIPTION
**Pull request by haircommander**
#### What type of PR is this?

 /kind documentation

#### What this PR does / why we need it:
including reordering some of the contents, and specify how to run with ipv6 disabled

#### Which issue(s) this PR fixes:

fixes https://github.com/cri-o/cri-o/issues/3437

#### Does this PR introduce a user-facing change?

```release-note
none
```

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Comment by saschagrunert** [comment]
/hold
git-validation seems to fail…oh man this new CI reporting...

**Comment by umohnani8** [comment]
LGTM

**Comment by haircommander** [comment]
fixed

/hold cancel

File: contrib/cni/README.md [hunk 1]
```diff
@@ -5,11 +5,31 @@ directory just contains some example configurations that can be used as the
...
 example directories).
 
-To use these configurations, place them in `/etc/cni/net.d` (or the directory
-specified by `crio.network.network_dir` in your overwrite of `/etc/crio/crio.conf.d`).
+### Configuration Directory
 
-CRI-O will only choose the network that has been specified via
-`crio.network.cni_default_network`.
+By default, your CNI configurations are read from `/etc/cni/net.d`.
+This can be overwritten by specifying `crio.network.network_dir` in your overwrite of `/etc/crio/crio.conf.d`
```
Comment (mrunalp) [inline contrib/cni/README.md:11]: `override of`

File: contrib/cni/README.md [hunk 2]
```diff
@@ -5,11 +5,31 @@ directory just contains some example configurations that can be used as the
...
+
+Unless you have a specific networking configuration you'd like to use, we recommend installing either [10-crio-bridge.conf][dual-stack], or [11-crio-ipv4-bridge.conf][ipv4-only]
+
+By default, we install the dual stack version: [10-crio-bridge.conf][dual-stack]
+
+However, if you are installing on a node with ipv6 disabled (`sysctl net.ipv6.conf.default.disable_ipv6` and `sysctl net.ipv6.conf.all.disable_ipv6` == 0)
+then we recommend you install the ipv4 only version: [11-crio-ipv4-bridge.conf][ipv4-only]
+Otherwise, you'll run into an error similar to:
+```
+Interface vetha38a080a Mac doesn't match: ee:7b:4d:57:3a:d9 not found
```
Comment (mrunalp) [inline contrib/cni/README.md:24]: I would say that this is a CNI plugin issue.

Comment (haircommander) [reply contrib/cni/README.md:24]: I would agree, and there are other cases that cause it too, but I think the workaround is worth documenting

File: contrib/cni/README.md (hunk 1 above)
Comment (haircommander) [reply contrib/cni/README.md:11]: fixed

File: contrib/cni/README.md (hunk 2 above)
Comment (mrunalp) [reply contrib/cni/README.md:24]: Yeah, I am fine with that, but let's open an issue with ocicni/cni to track.

**Pull request merged**